### PR TITLE
mwdeploy: add rsync stuff

### PR DESCRIPTION
**Pull request by RhinosF1**
To do:
- [x]  add main page canary check
- [x] add sync of l10n files
- [x] exit with a non zero code if failed
- [x] fix mw13
- [x] change rsync to cp for internal ones
- [x] check_up on canary

### CODE REVIEW

File: modules/mediawiki/files/bin/deploy-mediawiki [hunk 1]
```diff
@@ -12,35 +12,55 @@ def run(args, start):
...
         exitcodes.append(os.system('sudo -u www-data php /srv/mediawiki/w/maintenance/rebuildLocalisationCache.php --wiki=loginwiki'))
     if args.gitinfo:
         exitcodes.append(os.system('sudo -u www-data php /srv/mediawiki/w/extensions/MirahezeMagic/maintenance/rebuildVersionCache.php --use-staging --wiki=loginwiki'))
     if args.servers == 'skip':
-        print('very good because I have not wrote the rsync client')
+        print('Sync skipped. Mediawiki deploy has not passed canary stage.')
+        sync = False
     elif args.servers == 'all':
-        print('This would deploy everywhere')
+        serverlist = ['mw8', 'mw9', 'mw10', 'mw12', mw13', 'mwtask1']
```
Comment (RhinosF1) [inline modules/mediawiki/files/bin/deploy-mediawiki:46]: mw13 needs a '

File: modules/mediawiki/files/bin/deploy-mediawiki [hunk 2]
```diff
@@ -12,35 +12,55 @@ def run(args, start):
...
         files = str(args.files).split(',')
         for file in files:
             exitcodes.append(os.system(f'sudo -u www-data cp /srv/mediawiki-staging/{file} /srv/mediawiki/{file}'))
+            rsyncfiles.append(file)
     if args.folders:
         folders = str(args.folders).split(',')
         for folder in folders:
             exitcodes.append(os.system(f'sudo -u www-data cp -r /srv/mediawiki-staging/{folder}/* /srv/mediawiki/{folder}/'))
+            rsyncpaths.append(f'/srv/mediawiki/{folder}')
     if args.l10n:
```
Comment (RhinosF1) [inline modules/mediawiki/files/bin/deploy-mediawiki:37]: Need to add these to rsyncfiles

File: modules/mediawiki/files/bin/deploy-mediawiki (hunk 1 above)
Comment (Universal-Omega) [reply modules/mediawiki/files/bin/deploy-mediawiki:46]: Also does test3 need added here or not?

Comment (RhinosF1) [reply modules/mediawiki/files/bin/deploy-mediawiki:46]: No, it's handled seperately to production or we'd have a nightmare when testing upgrades.

**Pull request merged**